samples/highcharts/demo/calendar-heatmap-month/demo.js

```diff
@@ -0,0 +1,293 @@
+        const yCoordinate = Math.floor((firstWeekday + day - 1) / 7);
+        const id = day;
+
+        // Get the corresponding temperature for the current day from the given array
+        const temperature = data[day - 1].temperature;
+
+        chartData.push({
+            x: xCoordinate,
+            y: 4 - yCoordinate,
+            value: temperature,
+            date: new Date(date).getTime(),
+            custom: {
+                monthDay: id
+            }
+        });
+    }
+
+    // Fill in the missing values when dataset is looped through.
+    const emptyTilesLast = lengthOfWeek - lastWeekday;
+    for (let emptyDay = 1; emptyDay <= emptyTilesLast; emptyDay++) {
+        chartData.push({
+            x: (lastWeekday + emptyDay) % 7,
+            y: 0,
+            value: null,
+            date: null,
+            custom: {
+                empty: true
+            }
+        });
+    }
+    return chartData;
+}
+const chartData = generateChartData(data);
+
+Highcharts.chart('container', {
+    chart: {
+        type: 'heatmap'
+    },
+
+    title: {
+        text: 'Mid-day temperature May 2017',
+        align: 'center'
+    },
+
+    subtitle: {
+        text: 'Temperature variation at mid-day through May',
+        align: 'center'
+    },
+
+    accessibility: {
+        landmarkVerbosity: 'one'
+    },
+
+    tooltip: {
+        enabled: true,
+        outside: true,
+        zIndex: 20,
+        headerFormat: '',
+        pointFormat: '{#unless point.custom.empty}{point.date:%A, %b %e, %Y}{/unless}',
+        nullFormat: 'No data'
+    },
+
+    xAxis: {
+        categories: weekdays,
+        opposite: true,
+        lineWidth: 55,
+        lineColor: '#BBBAC5',
+        labels: {
+            rotation: 0,
+            y: 6,
+            style: {
+                textTransform: 'uppercase',
+                fontWeight: 'bold'
+            }
+        },
+        accessibility: {
+            description: 'weekdays',
+            rangeDescription: 'X Axis is showing all 7 days of the week, starting with Sunday.'
+        }
+    },
+
+    yAxis: {
+        labels: {
+            enabled: false
+        },
+        title: {
+            enabled: false
+        },
+        min: 0,
+        max: 4,
+        accessibility: {
+            description: 'weeks'
+        }
+    },
+
+    legend: {
+        align: 'right',
+        layout: 'vertical',
+        verticalAlign: 'middle'
+    },
+
+    colorAxis: {
+        min: 0,
+        stops: [
+            [0.2, 'lightblue'],
+            [0.4, '#CBDFC8'],
+            [0.6, '#F3E99E'],
+            [0.9, '#F9A05C']
+        ],
+        labels: {
+            format: '{value} °C'
+        }
+    },
+
+    series: [{
+        keys: ['x', 'y', 'value', 'date', 'id'],
+        data: chartData,
+        nullColor: 'whitesmoke',
+        borderWidth: 2,
+        borderColor: 'whitesmoke',
```

Comment:
The whitesmoke color looks a bit too contrasted in dark designs (like BrandDark). A trick for neutral colors that work well with both light and dark designs, is to use a grey color with semi-opacity. For example `rgba(196,196,196,0.2)` seems to strike the right balance in this case.

As a consequence, the grid lines of the y-axis must be hidden, and the `lineWidth` for the x-axis adjusted.